### ISSUE TITLE
64-bit divisions

### ISSUE THREAD

**Issue by Darksonn** [rust-lang/rust-clippy#16509]
### What it does

If the user code performs a u64/u64 division, the lint is triggered.

### Advantage

On some 32-bit platforms it may be very expensive to perform a 64-bit division. A project may wish to have anyone who tries to do this think twice - possibly enforce that such divisions are carried out using an explicit function call rather than the `/` operator.

### Drawbacks

Niche use-case. Most would not turn it on.

### Example

```rust
10u64 / 2u64
```

Could be written as:

```rust
// I know this is expensive, but I need to do it anyway.
div_64bit(10, 2)
```

### Comparison with existing lints

This may be solved if either of these are closed:

* https://github.com/rust-lang/rust-clippy/issues/8581
* https://github.com/rust-lang/rust-clippy/issues/15765

assuming that `/` operator usage triggers above lint when `Div` trait or `<u64 as Div>::div` is marked as disallowed.

### Additional Context

Please see [the Linux kernel mailing list](https://lore.kernel.org/all/20260204130653.GH2995752@noisy.programming.kicks-ass.net/).

cc @ojeda
